tensorflow/c/tensor_shape_utils.h

```diff
@@ -0,0 +1,31 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
```

Comment:
Can you move these files to `tensorflow/c/kernels/` .  I think it would be confusing to have it in `tensorflow/c/` directory since that directory contains TF public C APIs. Instead this is just a helper for the kernels.

Can you also add a comment at the top of the file (right below license comment) that says this file contains shape utilities to be used by kernels and is not part of the C API. As such, it is subject to change at any time.